sunshine/confighttp.cpp

```diff
@@ -412,6 +417,59 @@ void deleteApp(resp_https_t response, req_https_t request) {
   proc::refresh(config::stream.file_apps);
 }
 
+void uploadCover(resp_https_t response, req_https_t request) {
+  if(!authenticate(response, request)) return;
+
+  std::stringstream ss;
+  std::stringstream configStream;
+  ss << request->content.rdbuf();
+  pt::ptree outputTree;
+  auto g = util::fail_guard([&]() {
+    std::ostringstream data;
+
+    SimpleWeb::StatusCode code = SimpleWeb::StatusCode::success_ok;
+    if (outputTree.get_child_optional("error").has_value()) {
+      code = SimpleWeb::StatusCode::client_error_bad_request;
+    }
+
+    pt::write_json(data, outputTree);
+    response->write(code, data.str());
+  });
+  pt::ptree inputTree;
+  try {
+    //TODO: Input Validation
+    pt::read_json(ss, inputTree);
+  }
+  catch(std::exception &e) {
+    BOOST_LOG(warning) << "UploadCover: "sv << e.what();
+    outputTree.put("status", "false");
+    outputTree.put("error", e.what());
+    return;
+  }
+
+  auto key = inputTree.get<std::string>("key");
+  auto url = inputTree.get("url", "");
+
+  const std::string coverdir = SUNSHINE_ASSETS_DIR "/covers/";
+  if(!boost::filesystem::exists(coverdir)) {
+    boost::filesystem::create_directory(coverdir);
+  }
+
+  std::basic_string path = coverdir + key + ".png";
+  if (!url.empty()) {
+    if (!http::download_file(url, path)) {
```

Comment:
Probably we should check here if the URL provided in input is actually a trusted one (as if, from IGDB or similar)

Reply:
Good suggestion! Now only `images.igdb.com` is allowed. Also, `key` will be percent encoded to prevent special chars.